### DISCUSSION TITLE
Postasharp 2023 and Metalama RC compatibility

### DISCUSSION THREAD

**Discussion by omiguez** [metalama/Metalama#89] [Q&A] [answered]
Hello.

This is a followup of [this old question of mine.](https://github.com/postsharp/Metalama/discussions/9)

In the PostSharp 2023 announcement, you specify that it is not recommended to use PostSharp and Metalama in the same project. Does this apply to custom aspect creation, or aspect application, or both? We have a core project where we define all our aspects, and those aspects are used basically everywhere else. We'd like to migrate aspects one by one if possible.

Would it work if we split the core project into two, one with the aspects migrated to Metalama, and the other with the old aspects that have yet to migrated, add a reference from one of them into the other, and have the rest of our code reference the last one?
As in CommonAspects.Metalama ->CommonAspects.OldPostSharp->Any other project that needs to apply the aspects.

Or would we need to migrate all aspects at once?

The common aspects project is referenced both as a project and as a NuGet, depending on the solution.

**Comment by gfraiteur** [comment, marked as answer]
Hello,

This question is discussed at length [here](https://doc.metalama.net/migration/when-migrate).

We recommend indeed that you migrate all aspects at once. Migrating aspects one by one may work but you are likely going to bump into some issues and currently we haven't done anything to improve or support this scenario.
